### PR TITLE
Functional tests

### PR DESCRIPTION
**Pull request by anthonycr**
#### Summary
The purpose of this PR is to increase test coverage and enhance understandability of the library. The PR accomplishes this by:
- Moving all example models and tests not relevant to the sample app into `integration-test-` modules.
- Renaming all non sample-app samples to `integration-test-`.
- Adding unit tests to verify type adapter generation.
- Adding functional tests to verify compilation with the `StagProcessor`.

#### How to Test
CI should run all the unit tests.

### CODE REVIEW

**Comment by anthonycr** [comment]
cc @anirudhramanan @yasirmhd in case this affects your WIP PR.

**Comment by yasirmhd** [comment]
@anthonycr thanks for the heads up. We are almost done on our end and we will update the PR soon. You can go ahead and merge this. We will merge it back

File: .travis.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ jdk:
 android:
   components:
     - tools
-    - build-tools-26.0.2
-    - android-26
+    - build-tools-27.0.1
```
Comment (msya) [inline .travis.yml]: This could be updated to 27.0.2.

Comment (anthonycr) [reply .travis.yml]: 27.0.1 is the latest version, but I have updated the support library version used by the sample to 27.0.2 (that is the latest support lib version).

**Pull request merged**